contracts/src/UniswapV3MultiQuoter.sol

```diff
@@ -295,9 +325,17 @@ contract UniswapV3MultiQuoter {
             (result.amounts0[i], result.amounts1[i]) = zeroForOne == exactInput
                 ? (amounts[i] - state.amountSpecifiedRemaining, state.amountCalculated)
                 : (state.amountCalculated, amounts[i] - state.amountSpecifiedRemaining);
+            result.gasEstimates[i] = scaleMultiswapGasEstimate(state.gasAggregate);
         }
+    }
 
-        result.gasEstimates[state.amountsIndex] = gasBefore - gasleft();
+    /// @notice Returns the multiswap gas estimate scaled to UniswapV3Pool:swap estimates
+    /// @dev These parameters have been determined by running a linear regression between UniswapV3QuoterV2
+    /// gas estimates and the unscaled gas estimates from multiswap
+    /// @param multiSwapEstimate the gas estimate from multiswap
+    /// @return swapEstimate the gas estimate equivalent for UniswapV3Pool:swap
+    function scaleMultiswapGasEstimate(uint256 multiSwapEstimate) private view returns (uint256 swapEstimate) {
```

Comment:
can we include the test data and how the linear regression parameters were estimated somewhere in the repo? we'll probably have to do something similar for the new AMMs